web/packages/teleterm/src/ui/StatusBar/StatusBar.tsx

```diff
@@ -35,18 +37,47 @@ export function StatusBar() {
       alignItems="center"
       justifyContent="space-between"
       px={2}
+      gap={2}
+      color="text.slightlyMuted"
       overflow="hidden"
     >
-      <Text
-        color="text.slightlyMuted"
-        fontSize="14px"
+      <Flex
         css={`
-          white-space: nowrap;
+          // If the breadcrumbs are wider than the available space,
+          // allow scrolling them horizontally, but do not show the scrollbar.
+          width: 100%;
+          overflow: scroll;
+
+          &::-webkit-scrollbar {
+            display: none;
+          }
         `}
-        title={clusterBreadcrumbs && `Current cluster: ${clusterBreadcrumbs}`}
       >
-        {clusterBreadcrumbs}
-      </Text>
+        {breadcrumbs && (
+          <Flex
+            aria-label="breadcrumbs"
+            gap={2}
+            css={`
+              flex-shrink: 0;
+              font-size: 13px;
+            `}
+            title="Active tab"
+          >
+            {breadcrumbs.map((breadcrumb, index) => (
+              <Fragment key={breadcrumbs.slice(0, index).join('-')}>
+                {breadcrumb.Icon && (
+                  <breadcrumb.Icon color="text.muted" size="small" mr={-1} />
+                )}
+                <Text>{breadcrumb.name}</Text>
+                {index !== breadcrumbs.length - 1 && (
+                  <Text color="text.disabled">→</Text>
+                )}
+              </Fragment>
```

Comment:
wouldnt it be cool if React had some sort of thing like `String.join` but with components instead? And instead of appending some character at the end, it took an optional React node? That'd be sick

Reply:
I thought exactly the same thing when I was coding this!